src/mantle/framework/console/generators/trait-with-psr-4-file.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Has_PSR_4_Folder_Path trait file
+ *
+ * @package Mantle
+ */
+
+namespace Mantle\Framework\Console\Generators;
+
+use function Mantle\Support\Helpers\str;
+
+/**
+ * Generator support for PSR-4 files.
+ *
+ * @mixin \Mantle\Framework\Console\Generators\Generator_Command
+ */
+trait With_PSR_4_File {
+	/**
+	 * Get the class name to use.
+	 *
+	 * @param string $name Inputted name.
+	 * @return string
+	 */
+	protected function get_class_name( string $name ): string {
+		$name = str( str( $name )->explode( '\\' )->pop() )->studly();
+
+		if ( 'Tests' === $this->type && ! str( $name )->endsWith( 'Test' ) ) {
+			$name = $name->append( 'Test' );
+		}
+
+		return $name->value();
+	}
+
+	/**
+	 * Get the folder location of the file.
+	 *
+	 * @param string $name Name to use.
+	 * @return string
+	 */
+	protected function get_folder_path( string $name ): string {
+		$parts = str( $name )->explode( '\\' );
+
+		$parts->pop();
+
+		$folder = $parts
+			->filter()
```

Comment:
What does filter with no args do?

Reply:
runs `array_filter()` over the array to remove anything null